sdk/python/tests/dsl/io_types_test.py

```diff
@@ -0,0 +1,51 @@
+# Copyright 2021 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Tests for kfp.dsl.io_types."""
+
+import unittest
+import json
+import os
+
+from kfp.dsl import io_types
+
+
+class IOTypesTest(unittest.TestCase):
+
+  def test_complex_metrics(self):
+    metrics = io_types.ClassificationMetrics()
+    metrics.log_roc_reading(0.1, 98.2, 96.2)
+    metrics.log_roc_reading(24.3, 24.5, 98.4)
+    metrics.set_confusion_matrix_categories(['dog', 'cat', 'horses'])
+    metrics.log_confusion_matrix_row('dog', [2, 6, 0])
+    metrics.log_confusion_matrix_cell('cat', 'dog', 3)
+    metrics.metadata["test"] = 1.0
+    with open(os.path.join(os.path.dirname(__file__),
```

Comment:
since you're comparing JSON, can you just compare inline without the need for the test golden file?

Reply:
I am assuming you mean have the json string inline. Apart from the fact that the json string is quite large it will be easier to have it well indented in a testdata file rather then inline. Let me know if you strongly feel otherwise will move it inline.